pulser/waveforms.py

```diff
@@ -17,6 +17,7 @@
 
 from abc import ABC, abstractmethod
 import functools
+from functools import cached_property
```

Comment:
Let's make this import conditional on the version, like was done with `typing-extensions`. If in 3.7,   do `from cached_property import cached_property` instead.

Reply:
On second-thought, use `backports.cached_property` instead.

Reply:
:+1: